MotionPhotoReader/app/src/main/java/com/google/android/libraries/motionphotoreader/MotionPhotoReader.java

```diff
@@ -278,17 +271,61 @@ public void onOutputFormatChanged(@NonNull MediaCodec codec,
         if (surface != null) {
             outputSurface = new OutputSurface(renderHandler, motionPhotoInfo);
             outputSurface.setSurface(surface, surfaceWidth, surfaceHeight);
-            lowResDecoder.configure(videoFormat, outputSurface.getDecodeSurface(), null, 0);
+            decoder.configure(videoFormat, outputSurface.getDecodeSurface(), null, 0);
         } else {
-            lowResDecoder.configure(videoFormat, null, null, 0);
+            decoder.configure(videoFormat, null, null, 0);
+        }
+        decoder.start();
+
+        // Configure the buffer processor
+        bufferProcessor.configure(outputSurface, extractor, decoder, stabilizationOn);
+    }
+
+    /**
+     * Determines whether the motion photo is pre-stabilized, in which case we should not stabilize
+     * the video.
+     * @throws com.google.protobuf.InvalidProtocolBufferException if parsing an invalid proto object
+     */
+    private boolean isStabilizationOn(MotionPhotoInfo motionPhotoInfo, boolean stabilizationOn)
```

Comment:
can we call this isAlreadyStabilized ?  also, the method doesn't need to know if stabilization is enabled or not. That logic should be taken care of by the caller.